src/core.ts

```diff
@@ -175,7 +178,14 @@ export function getTimestamp(unix?: boolean, raw?: boolean) {
 
 export function instrument(eventState: IInstrumentationEventState) {
   if (config.instrument) {
-    addEvent({type: InstrumentationEventName, state: eventState});
+    addEvent({ type: InstrumentationEventName, state: eventState });
+  }
+}
+
+export function mark(key: string, info: any) {
```

Comment:
>any [](start = 40, length = 3)

Might want to require this to be a property bag, not allow any JS type.

Reply:
Since this is custom instrumentation, i don't think we can dictate what format payload should have. We can make these events searchable by keys on the backend, but it'd be up to webmaster to write/read payload later on

---
In reply to: [250283112](https://github.com/Microsoft/clarity-js/pull/113#discussion_r250283112) [](ancestors = 250283112)